### PR TITLE
OP-373 - Added Sylius 1.12 & 1.13 Support

### PR DESCRIPTION
**Pull request by tomkalon**
Added Sylius 1.12 & 1.13 Support
+ Gulp -> Webpack
+ dispatcher problem resolved
+ PHPSpec & PHPUnit has been refactored
+ ShippingExportEventListener has been refactored 
(ExportShipmentEven::class has been replaced by ResourceControllerEvent::class).

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -18,32 +18,23 @@ jobs:
...
-            php: 7.3
-          -
-            sylius: ~1.11.0
-            php: 7.3
-          -
-            sylius: ~1.11.0
-            php: 7.4
-
+          - sylius: "~1.13.0"
+            php: "8.0"
```
Comment (senghe) [inline .github/workflows/build.yml:29]: But you have no 8.0 in your matrix...

File: src/EventListener/ShippingExportEventListener.php [hunk 2]
```diff
@@ -90,16 +80,21 @@ public function exportShipment(ExportShipmentEvent $exportShipmentEvent): void
         } catch (ByrdApiException $e) {
             $shippingExport->setState('failed');
             $this->entityManager->flush();
-
-            $exportShipmentEvent->addErrorFlash(
-                sprintf('Byrd error for order %s: %s', $order->getNumber(), $e->getMessage())
+            $this->requestStack->getSession()->getBag('flashes')->add(
```
Comment (senghe) [inline src/EventListener/ShippingExportEventListener.php:83]: You could put the flashbag to a variable

File: src/EventListener/ShippingExportEventListener.php [hunk 3]
```diff
@@ -135,15 +130,14 @@ public function autoExport(PaymentInterface $payment): void
...
 
-        $event = new ExportShipmentEvent(
-            $exportObject,
-            $this->flashBag,
-            $this->entityManager,
-            $this->filesystem,
-            $this->translator,
-            $this->shippingLabelsPath
-        );
+        $exportShipmentEvent = new ResourceControllerEvent($exportObject);
```
Comment (senghe) [inline src/EventListener/ShippingExportEventListener.php:133]: Why this change?

File: tests/Application/.babelrc
Comment (senghe) [inline tests/Application/.babelrc:1]: Why this file has been removed?

File: tests/Application/config/packages/webpackl_encore.yml
Comment (senghe) [inline tests/Application/config/packages/webpackl_encore.yml:1]: Typo in file name

File: tests/Application/config/packages/webpackl_encore.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+webpack_encore:
+    output_path: '%kernel.project_dir%/public/build/default'
+    builds:
+        shop: '%kernel.project_dir%/public/build/shop'
```
Comment (senghe) [inline tests/Application/config/packages/webpackl_encore.yml:4]: why in assets.yaml we have 4 positions, but here we have only two? As I remember, the counts usually are equal

**Review by senghe — CHANGES_REQUESTED** [review]

File: src/EventListener/ShippingExportEventListener.php (hunk 3 above)
Comment (tomkalon) [reply src/EventListener/ShippingExportEventListener.php:133]: `The ExportShipmentEvent is deprecated since Sylius 1.8 and will be removed. Use \Sylius\Bundle\ResourceBundle\Event\ResourceControllerEvent instead`

File: tests/Application/.babelrc
Comment (tomkalon) [reply tests/Application/.babelrc:1]: https://docs.sylius.com/en/1.12/cookbook/frontend/migrating-to-webpack-1-12-or-later.html

> 2. Remove the following files from you project:
> 
> .babelrc
> gulpfile.babel.js
> yarn.lock
> public/assets
> node_modules

File: .github/workflows/build.yml (hunk 1 above)
Comment (tomkalon) [reply .github/workflows/build.yml:29]: - Fixed

File: src/EventListener/ShippingExportEventListener.php (hunk 2 above)
Comment (tomkalon) [reply src/EventListener/ShippingExportEventListener.php]: Fixed

File: tests/Application/config/packages/webpackl_encore.yml
Comment (tomkalon) [reply tests/Application/config/packages/webpackl_encore.yml:1]: Fixed

File: tests/Application/config/packages/webpackl_encore.yml (hunk 4 above)
Comment (tomkalon) [reply tests/Application/config/packages/webpackl_encore.yml]: Fixed

**Review by senghe — APPROVED** [review]
